app/Models/Campaign.php

```diff
@@ -110,6 +110,17 @@ public function scopeWhereClosed($query)
             });
     }
 
+    /**
+     * Scope a query to only include campaigns containing specified cause within
+     * "cause" field.
+     */
+    public function scopeWhereHasCause($query, $cause)
+    {
+        // Regex matches on comma separated words to ensure precise match excepting first
+        // and last words (matching on only a following or preceding comma).
```

Comment:
I might be reading this wrong but I'm confused - can you explain to me what exactly we're doing here?

Reply:
I basically used the comma separation as 'word boundaries', so that we'd get precise matches (so the `filter[has_cause]=rights` wouldn't yield `lgbtq-rights` & `gender-rights`). The exceptions are the first and last causes, which do not have a preceding comma (first) or following comma (last), so in those case the regex only looks for the one comma they _do_ have.

Does this make sense? You can play with the regex matching [here](https://regex101.com/r/D1dRxd/6) to see what I mean.

D'you think this comment could use a little re-writing for clarity?

Reply:
AH that sandbox thing really helped - is this summary correct? We look at each comma-separated cause, and signal a match if we have a complete match on one of the comma-separated causes?

Reply:
I'm not sure how long that link sticks around, but if it's a forever-thing it'd be nice to include in the comment! I think seeing it actually visualized like that (and being able to type in different example inputs) is super helpful.

Reply:
>  is this summary correct? We look at each comma-separated cause, and signal a match if we have a complete match on one of the comma-separated causes?

Precisely so!

> I'm not sure how long that link sticks around, but if it's a forever-thing it'd be nice to include in the comment! I think seeing it actually visualized like that (and being able to type in different example inputs) is super helpful.

That's a really good idea! I'll look into that

Reply:
Ahhh okay the comma stuff had me confused - I took `matches on comma separated words` to include the fact that the first word would not have a comma before it and same idea for last word. I think this would have been more clear for me:
```
Regex matches on comma separated words to ensure precise match.  Accounts for first and last words only having a comma on one side.
```
And I love the idea of including that link if it's forever!

Reply:
Simplified the comment and added a link to the regex sandbox in https://github.com/DoSomething/rogue/pull/985/commits/3c1db33a7ffa7710d08138a52dcdacde12d4f420

Reply:
Oh whoops I missed your suggestion @katiecrane I like that clarification better!

Reply:
love it!

Reply:
oh lol either way!

Reply:
Updated in https://github.com/DoSomething/rogue/pull/985/commits/393e1f7d9ffe106bd6d16885d39cc5281da6d3cf